packages/jest-jasmine2/src/setup-jest-globals.js

```diff
@@ -48,59 +46,24 @@ const addSuppressedErrors = result => {
   }
 };
 
-function addAssertionErrors(result) {
-  const {
-    assertionCalls,
-    expectedAssertionsNumber,
-    isExpectingAssertions,
-  } = getState();
-  setState({
-    assertionCalls: 0,
-    expectedAssertionsNumber: null,
-  });
-  if (
-    typeof expectedAssertionsNumber === 'number' &&
-    assertionCalls !== expectedAssertionsNumber
-  ) {
-    const expected = EXPECTED_COLOR(
-      pluralize('assertion', expectedAssertionsNumber),
-    );
-    const message = new Error(
-      matcherHint('.assertions', '', String(expectedAssertionsNumber), {
-        isDirectExpectCall: true,
-      }) +
-        '\n\n' +
-        `Expected ${expected} to be called but only received ` +
-        RECEIVED_COLOR(pluralize('assertion call', assertionCalls || 0)) +
-        '.',
-    ).stack;
-    result.status = 'failed';
-    result.failedExpectations.push({
-      actual: assertionCalls,
-      expected: expectedAssertionsNumber,
-      message,
-      passed: false,
+const addAssertionErrors = result => {
+  const assertionErrors = extractExpectedAssertionsErrors();
+  if (assertionErrors.length) {
+    const jasmineErrors = assertionErrors.map(({actual, error, expected}) => {
+      return {
+        actual,
+        expected,
+        message: error.stack,
+        passed: false,
+      };
     });
-  }
-  if (isExpectingAssertions && assertionCalls === 0) {
-    const expected = EXPECTED_COLOR('at least one assertion');
-    const received = RECEIVED_COLOR('received none');
-    const message = new Error(
-      matcherHint('.hasAssertions', '', '', {
-        isDirectExpectCall: true,
-      }) +
-        '\n\n' +
-        `Expected ${expected} to be called but ${received}.`,
-    ).stack;
     result.status = 'failed';
-    result.failedExpectations.push({
-      actual: 'none',
-      expected: 'at least one',
-      message,
-      passed: false,
-    });
+    result.failedExpectations = [
+      ...result.failedExpectations,
+      ...jasmineErrors,
```

Comment:
btw, this kinda always pissed me up, here's PR with a fix: https://github.com/facebook/jest/pull/3876